modules/internal/DefaultRectangular.chpl

```diff
@@ -1036,13 +1036,24 @@ module DefaultRectangular {
     }
 
     proc dsiReallocate(d: domain) {
+      on this {
+      //
+      // If both d and dom are default rectangular, this is pretty
+      // easy...
+      //
       if (d._value.type == dom.type) {
-        on this {
         var copy = new DefaultRectangularArr(eltType=eltType, rank=rank,
-                                            idxType=idxType,
-                                            stridable=d._value.stridable,
-                                            dom=d._value);
-        for i in d((...dom.ranges)) do
+                                             idxType=idxType,
+                                             stridable=d._value.stridable,
+                                             dom=d._value);
+        //
+        // TODO: Making this for into a forall ought to accelerate
+        // dsiReallocate() calls, yet doing so breaks due to uint/int
+        // interaction issues today.  Deserves more of a look...
+        // Does our standalone parallel iterator not have the same
+        // type flexibility as the serial iterator?
+        //
+        for i in d[(...dom.ranges)] do
```

Comment:
I don't plan on doing a full review, but it might be interesting to see if `forall i in zip(d[(...dom.ranges)]) do` works. Just using `zip` will trigger l/f so it could at least narrow down if the standalone iter is what's broken.

Also, I think it'd be nice if there was a JIRA issue to track this.